apps/src/templates/projects/projectsRedux.js

```diff
@@ -104,6 +105,18 @@ function personalProjectsList(state = initialPersonalProjectsList, action) {
         ...state,
         projects: updatedProjects
       };
+    case UNPUBLISH_SUCCESS:
+      var unpublishedChannel = action.projectId;
+
+      var unpublishedProjectIndex = state.projects.findIndex(project => project.channel === unpublishedChannel);
+
+      var newProjects = [...state.projects];
+      newProjects[unpublishedProjectIndex].publishedAt = null;
```

Comment:
When we start allowing you to share from the quick actions menu, unpublishing from within the share dialog could cause this case in this reducer to fire, giving you an error when trying to set `newProjects[-1].publishedAt`. 

At the very least you should check that `unpublishedProjectIndex` is a valid array index before trying to set `publishedAt`. A better approach would be to do something like:
```js
this.props.unpublishProject(this.props.projectId).then(dispatch(markUnpublished(projectId)))
```
from within PersonalProjectsTableActionsCell.

Reply:
thinking on this a bit more, I'm not sure it makes sense to use the shareDialogRedux action `unpublishProject` here, because even if you use `.then` as suggested above, it still allows for the action types to leak in the opposite direction if you were to unpublish the project and then open the share dialog before the unpublish had completed. 

I would recommend instead issuing the unpublish XHR separately within this reducer, using its own action types, so that this component can be completely isolated from the share dialog.

Reply:
I isolated unpublishing from the share dialog. @davidsbailey Do the updates look like what you were thinking?

Reply:
looks good!